### PR TITLE
fix: change $defs to definitions per draft07

### PR DESCRIPTION
**Pull request by toddbaert**
`$defs` works, but it its not part of draft07 which leads to odd warnings in some implementations.

I've switched this to `definitions` as per the spec.

see: https://json-schema.org/draft-07/draft-handrews-json-schema-validation-01#rfc.section.9

fixes: https://github.com/open-feature/flagd/issues/1309

### CODE REVIEW

File: json/flags.yaml [hunk 1]
```diff
@@ -99,20 +99,20 @@ properties:
         patternProperties:
           "^.{1,}$":
             type: object
-  $comment: merge the variants with the base flag to build our typed flags
   booleanFlag:
+    $comment: merge the variants with the base flag to build our typed flags
```
Comment (toddbaert) [inline json/flags.yaml:103]: This is the only unrelated change... for some reason changing `$defs` to `definitions` resulted in saying the location of this commend was previously invalid.

**Review by thisthat — APPROVED** [review]

**Review by beeme1mr — APPROVED** [review]

**Review by Kavindu-Dodan — APPROVED** [review]
